File: tests/examples_tests/python/wrappers/classes/room_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "examples/cxx/classes/room.hpp"

namespace py = pybind11;

void bindRoom(py::module& m) {
    using namespace iegen::example;
    py::class_<iegen::example::Room> room(m, "Room");
room.def("calculate_area", &iegen::example::Room::calculateArea);
    room.def("calculate_volume", &iegen::example::Room::calculateVolume);
}

File: tests/examples_tests/python/src/classes/room_pygen.py
from __future__ import annotations
from typing import *
import iegen.examples.classes.room as pybind_room_pygen
from bind_utils import *
from metaclass import *


class Room(metaclass=OriginalMethodsMetaclass):
    """
    comments
    """
    
    
    @bind
    def calculate_area(self) -> float:
        """
        comments
        """
        pass
    @bind
    def calculate_volume(self) -> float:
        """
        comments
        """
        pass

